sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/route/inline/InlineShardingStrategy.java

```diff
@@ -58,7 +61,17 @@ public InlineShardingStrategy(final InlineShardingStrategyConfiguration inlineSh
         Preconditions.checkState(shardingValue instanceof ListRouteValue, "Inline strategy cannot support range sharding.");
         Collection<String> shardingResult = doSharding((ListRouteValue) shardingValue);
         Collection<String> result = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
-        result.addAll(shardingResult);
+        Map<String, String> availableTargetNamesMap = Maps.uniqueIndex(availableTargetNames, new Function<String, String>() {
+            @Override
+            public String apply(String input) {
+                return input;
+            }
+        });
```

Comment:
This part may be unnecessary, because availableTargetNames must come from `TableRule`, and `TableRule` must return `LinkedHashSet`.